src/rest_framework_jwt/utils.py

```diff
@@ -95,8 +95,23 @@ def jwt_get_username_from_payload_handler(payload):
 def jwt_encode_payload(payload):
     """Encode JWT token claims."""
 
-    key = api_settings.JWT_PRIVATE_KEY or jwt_get_secret_key(payload)
-    return force_text(jwt.encode(payload, key, api_settings.JWT_ALGORITHM))
+    headers=None
+
+    algo = api_settings.JWT_ALGORITHM
+    if isinstance(algo,list):
+        algo = algo[0]
+    if algo.startswith("HS"):
+        key = jwt_get_secret_key(payload)
+    else:
+        key = api_settings.JWT_PRIVATE_KEY
+
+    if isinstance(key, dict):
+        kid, key = next(iter(key.items()))
+        headers = {"kid": kid}
+    elif isinstance(key,list):
+        key = key[0]
```

Comment:
I would suggest splitting signing keys from verification keys to make it as explicit as possible. I would also suggest using `signing_key`.

Reply:
see top level discussion for suggestions, I'd appreciate your feedback